teams.md/docs/csharp/in-depth-guides/user-authentication/README.md

```diff
@@ -1,22 +1,104 @@
 ---
 sidebar_position: 4
-summary: Enable user authentication in Teams apps to access secured resources like email, flight status, and online services.
+summary: API guide in C# to implement User Authentication with SSO in Teams Apps.
 ---
 
 # 🔒 User Authentication
 
-<!-- 
-Things to potentially add to this section:
+Once you have configured your Azure Bot resource and OAuth settings, as described in [User Authentication Setup](/teams/user-authentication/sso-setup), add the following code to your `App`:
 
-- The name of the auth is fixed to `graph` here, but it can easily be changed by supplying a value when building the App.
 
-- Show that for explicit oauth you can configure the oauth card that is sent to the user via the options to the signin function.
+## Configure the OAuth connection
 
-- Create mermaid diagrams for how sso and oauth works
---->
+```cs
+var builder = WebApplication.CreateBuilder(args);
 
-At times agents must access secured online resources on behalf of the user, such as checking email, checking on flight status, or placing an order. To enable this, the user must authenticate their identity and grant consent for the application to access these resources. This process results in the application receiving a token, which the application can then use to access the permitted resources on the user's behalf.
+var appBuilder = App.Builder()
+    .AddOAuth("graph");
 
-## Resources
+builder.AddTeams(appBuilder);
+var app = builder.Build();
+var teams = app.UseTeams();
+```
+:::tip
+Make sure you use the same name you used when creating the OAuth connection in the Azure Bot Service resource.
+:::
+
+## Signing In
+
+You must call the `signin` method inside your route handler, for example: to signin when receiving the `/signin` message:
+
+```cs
+teams.OnMessage("/signin", async context =>
```

Comment:
for these snippets do you think it's a good idea to also add a controller example along with the "minimal" example? ex. https://microsoft.github.io/teams-ai/csharp/essentials/on-activity/

<img width="1547" height="525" alt="image" src="https://github.com/user-attachments/assets/fe9f3668-0037-4ef7-9315-1d51738e9ef0" />